### ISSUE TITLE
There are now three ovirt modules. Is there a path forward?

### ISSUE THREAD

**Issue by ansibot** [ansible/ansible#29370]
From @wenottingham on 2016-09-28T16:22:54Z
##### ISSUE TYPE
- Bug Report
##### COMPONENT NAME

cloud/misc/ovirt.py
cloud/misc/rhevm.py (new in 2.2)
cloud/ovirt/ovirt_vms.py (new in 2.2)

##### ANSIBLE VERSION
2.2

##### SUMMARY

There are now three modules for the same thing in extras. Ideally, there's some sort of path forward to a single module.

CC @TimothyVandenbrande @machacekondra @vincentvdk 

Copied from original issue: ansible/ansible-modules-extras#3059

**Comment by ansibot** [comment]
From @ansibot on 2016-09-28T16:22:54Z



@joshainglis, @karmab, ping. This issue is waiting on your response.
[click here for bot help](https://github.com/ansible/ansibullbot/blob/master/ISSUE_HELP.md)

**Comment by ansibot** [comment]
From @machacekondra on 2016-09-28T16:22:54Z

ovirt_vms module is for version 4 of the oVirt. The other modules are for version 3 of the oVirt.

**Comment by ansibot** [comment]
From @TimothyVandenbrande on 2016-09-28T16:22:54Z

I expressed my concerns here: https://github.com/ansible/ansible/pull/17364#issuecomment-245210349
But yes, I think it would be better to get everything in 1 module and make the v4 api backwards compatible than truly maintaining 3. But yes, this also means you'll need the features of all 3 modules merged in 1. Maintaining 3 of them will prolly be required for the next releases, but we can put 2 on depricated for a while before decommisioning.
My most important feature would be the capability to create hosts with multiple nics and disks from that host or group's vars.
(and ofc the basic stuff as start/stop, VM options manipulation.)

The main reason I also wouldn't drop the v3 option is because as long as Red Hat supports it, you should be able to let people use it. (Especially since the latest RHEV/RHV version is still 3.6)

**Comment by ansibot** [comment]
From @machacekondra on 2016-09-28T16:22:54Z

There is still no decision when v3 API and v3 SDKs will stopped be supported, maybe it will be 4.1, maybe later, but for sure - it will be stopped supported and not backward compatible.

**Comment by ansibot** [comment]
From @vincentvdk on 2016-09-28T16:22:54Z

I've not managed an Ovirt/RHEV setup in some years now. So that's why I couldn't develop and test new features proposed for the original module. My idea was to split this up in a module per 'service' (vm, disk. dc, network, ...). I believe this is also the idea for the v4.x modules (yay! :-) ). 
As there will still be people who keep running v3, the v3 modules should not go away. But merging the 2 v3 modules might be preferable to not cause confusion.
Just my 2c.

**Comment by ansibot** [comment]
From @karmab on 2016-09-28T16:22:54Z

looking at the code , the cloud/ovirt/ovirt_vms.py module seems the more accurate and legible.
additionally, it uses ovirtsdk4 which seems to be the way to go in the future.
so i would tag the other ones as deprecated  or not recommended .
the rhevm.py is too complicated to read, plus i think the naming just adds confusion
a couple of notes on ovirt_vms.py
- why not call it ovirt_vm ....
- we should make sure it also works for ovirt3 installations

**Comment by ansibot** [comment]
From @machacekondra on 2016-09-28T16:22:54Z

> why not call it ovirt_vm ....

Well, maybe if it will be named _ovirt_vm_ someone will ask why not call it _ovirt_vms_. I didn't find any recomendation about naming. But no problem to rename it if more people will suggest it.

> we should make sure it also works for ovirt3 installations

I think it don't worth the effort. Someday in future ovirt3 won't be supported and we should remove that code again. So I would leave those current modules (and maybe rename them to ovirt3.py / rhevm3.py ) and deprecate them once oVirt 3 won't be supported anymore.

**Comment by ansibot** [comment]
From @wenottingham on 2016-09-28T16:22:54Z

OK, so here's what I think might be useful as 'fixes' to come out of this:
- [ ] Fix location/categorization such that these all end up in the same place in the docs (two are in 'cloud/misc', the rest in 'cloud/ovirt'
- [ ] Update description and module docs to more clearly define what versions of oVirt (and RHEV) they work to manage

Mostly just want to avoid drive-by confusion (like I had). Comments?

**Comment by ansibot** [comment]
@ansibot Greetings! Thanks for taking the time to open this issue. In order for the community to handle your issue effectively, we need a bit more information. 

Here are the items we could not find in your description:
- ansible version

Please set the description of this issue with this template:
https://raw.githubusercontent.com/ansible/ansible/devel/.github/ISSUE_TEMPLATE.md

[click here for bot help](https://github.com/ansible/ansibullbot/blob/master/ISSUE_HELP.md)

**Comment by ansibot** [comment]
cc @mwperina
[click here for bot help](https://github.com/ansible/ansibullbot/blob/master/ISSUE_HELP.md)

**Comment by ansibot** [comment]
cc @mnecas
[click here for bot help](https://github.com/ansible/ansibullbot/blob/master/ISSUE_HELP.md)

**Comment by Akasurde** [comment]
Appropriate information about migration and usage is added using collection [ovirt.ovirt](https://github.com/oVirt/ovirt-ansible-collection) and https://github.com/ansible-collections/community.general/pull/353.

Closing this as per above.
